tools/build_langserver/langserver_main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"gopkg.in/op/go-logging.v1"
+	"os"
+
+	"cli"
+	"github.com/sourcegraph/jsonrpc2"
+	"tools/build_langserver/langserver"
+)
+
+// TODO(bnmetrics): also think about how we can implement this with .build_defs as well
+// TODO(bnmetrics): Make the Usage part better
+var log = logging.MustGetLogger("build_langserver")
+
+var opts = struct {
+	Usage     string
+	Verbosity cli.Verbosity `short:"v" long:"verbosity" default:"warning" description:"Verbosity of output (higher number = more output)"`
+
+	Mode string `short:"m" long:"mode" default:"stdio" description:"Mode of the language server communication (stdio|tcp)"`
```

Comment:
FYI the flags thing can do this for you:
```
choice:"stdio" choice:"tcp" description:"Mode of the language server communication"
```
It'll enforce the appropriate values & generate nice help too.